tests/functional/codegen/features/test_assert.py

```diff
@@ -1,32 +1,28 @@
 import pytest
-from eth_tester.exceptions import TransactionFailed
 
 
 # web3 returns f"execution reverted: {err_str}"
 # TODO move exception string parsing logic into tx_failed
-def _fixup_err_str(s):
-    return s.replace("execution reverted: ", "")
+def _fixup_err_str(e_info):
+    return e_info.value.args[0].replace("execution reverted: ", "")
 
 
-def test_assert_refund(w3, get_contract_with_gas_estimation, tx_failed):
+def test_assert_refund(env, get_contract, tx_failed):
     code = """
 @external
 def foo():
     raise
     """
-    c = get_contract_with_gas_estimation(code)
-    a0 = w3.eth.accounts[0]
+    c = get_contract(code)
+    env.set_balance(env.deployer, 10**7)
     gas_sent = 10**6
-    tx_hash = c.foo(transact={"from": a0, "gas": gas_sent, "gasPrice": 10})
-    # More info on receipt status:
-    # https://github.com/ethereum/EIPs/blob/master/EIPS/eip-658.md#specification.
-    tx_receipt = w3.eth.get_transaction_receipt(tx_hash)
-    assert tx_receipt["status"] == 0
-    # Checks for gas refund from revert
-    assert tx_receipt["gasUsed"] < gas_sent
+    with tx_failed():
+        c.foo(gas=gas_sent, gas_price=10)
+
+    assert env.last_result.gas_used < gas_sent, "Gas refund not received"
```

Comment:
i don't think this the right error message. i think this is testing some really old behavior, before REVERT the only way to fail was to issue an invalid instruction which would consume all gas (see: https://eips.ethereum.org/EIPS/eip-140#motivation). after we added revert, we wanted to check that it was the "gas-friendly" revert, so we checked that gas_used < gas_sent
```suggestion
    # check we issued `revert`, which does not consume all gas
    assert env.last_result.gas_used < gas_sent, "Used all gas"
```